src/vmm/src/persist.rs

```diff
@@ -239,14 +241,18 @@ fn snapshot_state_to_file(
         .create(true)
         .write(true)
         .open(snapshot_path)
-        .map_err(SnapshotBackingFile)?;
+        .map_err(|e| SnapshotBackingFile("open", e))?;
 
     let mut snapshot = Snapshot::new(version_map, snapshot_data_version);
     snapshot
         .save(&mut snapshot_file, microvm_state)
         .map_err(SerializeMicrovmState)?;
-    snapshot_file.flush().map_err(SnapshotFileFlush)?;
-    snapshot_file.sync_all().map_err(SnapshotFileFlush)
+    snapshot_file
+        .flush()
+        .map_err(|e| SnapshotBackingFile("flush the content of", e))?;
```

Comment:
I would make the string more terse here. It can be just the name of the action (`flush`, `sync_all`). The resulting error would be for example: "Cannot perform flush on the snapshot backing file", "Cannot perform sync_all on the snapshot backing file", etc.

Reply:
@serban300 done, PTAL